src/Microsoft.Diagnostics.Runtime/Native/NativeHeap.cs

```diff
@@ -84,6 +154,121 @@ private void CreateFreeType()
             _types.Add(_free);
         }
 
+        public override ClrType GetClrTypeFromEE(ulong eeType)
+        {
+            if ((((int)eeType) & 3) != 0)
+                eeType &= ~3UL;
+
+            ClrType clrType = null;
+            int index;
+            if (_indices.TryGetValue(eeType, out index))
+                clrType = _types[index];
+            else
+                clrType = ConstructObjectType(eeType);
+
+            return clrType;
+        }
+
+        internal class NativeException : ClrException
```

Comment:
Can you move this class to its own file?  I realize the desktop one doesn't do that, but it'd be better...